src/app/zap-templates/zcl/data-model/chip/time-synchronization-cluster.xml

```diff
@@ -68,42 +101,81 @@ limitations under the License.
-    <attribute side="server" code="0x0005" define="TIME_ZONE" type="ARRAY" entryType="TimeZoneStruct" min="1" max="2" writable="true" optional="true">
-	<description>TimeZone</description>
-	<access op="read" privilege="view"/>
-	<access op="write" privilege="manage"/>
-    </attribute>
-    <attribute side="server" code="0x0006" define="DST_OFFSET" type="ARRAY" entryType="DSTOffsetStruct" max="20" writable="true" optional="true">
-	<description>DSTOffset</description>
-	<access op="read" privilege="view"/>
-	<access op="write" privilege="manage"/>
-    </attribute>		
+    <attribute side="server" code="0x0003" define="TRUSTED_TIME_SOURCE" type="TrustedTimeSourceStruct" isNullable="true" optional="true">TrustedTimeSource</attribute>
+    <attribute side="server" code="0x0004" define="DEFAULT_NTP" type="CHAR_STRING" length="128" isNullable="true" optional="true">DefaultNTP</attribute>
+    <attribute side="server" code="0x0005" define="TIME_ZONE" type="ARRAY" entryType="TimeZoneStruct" optional="true">TimeZone</attribute>
+    <attribute side="server" code="0x0006" define="DST_OFFSET" type="ARRAY" entryType="DSTOffsetStruct" optional="true">DSTOffset</attribute>
     <attribute side="server" code="0x0007" define="LOCAL_TIME" type="epoch_us" default="0x00" isNullable="true" optional="true">LocalTime</attribute>
-    <attribute side="server" code="0x0008" define="TIME_ZONE_DATABASE" type="boolean" default="0" optional="true">TimeZoneDatabase</attribute>
-    <attribute side="server" code="0x0009" define="NTP_SERVER_PORT" type="INT16U" isNullable="true"  optional="true">NtpServerPort</attribute>
-
-    <!-- Test Commands -->
-    <command source="client" code="0x00" name="SetUtcTime" optional="false">
-	<description>Upon receipt of this command, the server MAY update its UTCTime Attribute to match the time specified in the command</description>
-	<arg name="UtcTime" type="epoch_us"/>	
-	<arg name="Granularity" type="GranularityEnum"/>	
-	<arg name="TimeSource" type="TimeSourceEnum" optional="true"/>
-    </command>	
+    <attribute side="server" code="0x0008" define="TIME_ZONE_DATABASE" type="TimeZoneDatabaseEnum" default="2" optional="true">TimeZoneDatabase</attribute>
+    <attribute side="server" code="0x0009" define="NTP_SERVER_AVAILABLE" type="boolean" optional="true">NTPServerAvailable</attribute>
+    <attribute side="server" code="0x000A" define="TIME_ZONE_LIST_MAX_SIZE" type="INT8U" min="1" max="2" optional="true">TimeZoneListMaxSize</attribute>
+    <attribute side="server" code="0x000B" define="DST_OFFSET_LIST_MAX_SIZE" type="INT8U" min="1" optional="true">DSTOffsetListMaxSize</attribute>
+    <attribute side="server" code="0x000C" define="SUPPORTS_DNS_RESOLVE" type="boolean" optional="true">SupportsDNSResolve</attribute>
+
+    <command source="client" code="0x00" name="SetUTCTime" optional="false">
+      <description>This command MAY be issued by Administrator to set the time.</description>
+      <arg name="UTCTime" type="epoch_us"/>
```

Comment:
Properly also a stupid question due to a lack of insight into these files. But does command fields not use the optional="false" tag when mandatory? It seems to be used for other elements?

Reply:
It's implicit. The default is optional="false". May be it makes sense to explicitly declare it to avoid confusion.

Reply:
Ok, that was my assumption as well. The main thing that "confused" me was that it was explicitly stated in other elements so just wanted to make sure.